keras_core/backend/jax/distribute_test.py

```diff
@@ -0,0 +1,131 @@
+"""Tests for JAX based distribution."""
+import os
+
+import numpy as np
+import pytest
+
+from keras_core import backend
+from keras_core import layers
+from keras_core import models
+from keras_core import testing
+from keras_core.backend.common import distribute_scope
+from keras_core.backend.jax import distribute
+
+import jax
+
+# Config the number of devices for testing to be 8.
+flags = os.environ.get("XLA_FLAGS", "")
+os.environ["XLA_FLAGS"] = flags + " --xla_force_host_platform_device_count=8"
```

Comment:
This should presumably be run in a separate job since it will affect other tests?

Reply:
Good point. Let me see if I can move this to the setUp() since it can be done after importing the JAX.

Reply:
Somehow the setup module is not picked by the test, need to dig more.

Reply:
Contacted JAX team internally, seems to be a JAX/XLA issue for flag value parsing. Since the flag value is only parsed once, we can't really unset it afterwards. This means all the JAX tests are running with 8 CPUs at the moment (but will only use CPU:0 unless it is running for distribution)

We could split the distribution test into a separate pytest run in the action config, like applications.